doc/design/csp.md

```diff
@@ -109,52 +109,43 @@ fmt.Println("Received from channel")
 
 In Fluid, we should be able to perform the above operations on the channel objects as well. As of now, we support two different kinds of channels : [Buffered Channel](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/framework/details/buffered_channel.h) and [UnBuffered Channel](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/framework/details/unbuffered_channel.h)
 
-Send and Receive can be performed as following on a buffered channel:
+With a buffered channel, the send operation blocks if the buffer is full, and the receive operation blocks if the buffer is empty.  The following program should run without blocking because the amount of data written into the buffer is less than the buffer size.
 
 ```python
-import threading
-
-def send_to_channel(channel, num_time=1):
-  for i in xrange(num_time):
-    channel.send(i)
+import fluid
 
 # Create a buffered channel of capacity 10
-buffer_size = 10;
+buffer_size = 10
 ch = fluid.make_channel(dtype=INT, buffer_size)
 
 # Now write three elements to the channel
-thread = threading.Thread(target=send_to_channel, args=(ch, 3, ))
-thread.daemon = True
-thread.start()
+with fluid.while(steps=buffer_size):
+  fluid.send(ch, step)
```

Comment:
I think we can reuse these methods as `fluid.send()` and `fluid.receive()` for reading and writing in `select` cases as well.

Reply:
Maybe something like:
```
sel = fluid.select()
with sel.case(fluid.recv(ch), x): # similar to with sel.read_case(ch, x) or sel.case(ch.read(), x)
  do_something_with(x)
```